[32m2025-04-08 23:35:15 +0200[0m - dagster - [34mDEBUG[0m - __ASSET_JOB - 9b56ea6b-712d-414e-96b4-5071e368c4e9 - 440340 - LOGS_CAPTURED - Started capturing logs in process (pid: 440340).
[32m2025-04-08 23:35:15 +0200[0m - dagster - [34mDEBUG[0m - __ASSET_JOB - 9b56ea6b-712d-414e-96b4-5071e368c4e9 - 440340 - generate_pdf - STEP_START - Started execution of step "generate_pdf".
[32m2025-04-08 23:35:15 +0200[0m - dagster - [34mDEBUG[0m - __ASSET_JOB - 9b56ea6b-712d-414e-96b4-5071e368c4e9 - generate_pdf - Loading file from: /home/rafik/Bureau/Data_Engineering/challengeData/rafik_pipeline/.tmp_dagster_home_hp2rsqei/storage/calculRJ using PickledObjectFilesystemIOManager...
[32m2025-04-08 23:35:15 +0200[0m - dagster - [34mDEBUG[0m - __ASSET_JOB - 9b56ea6b-712d-414e-96b4-5071e368c4e9 - 440340 - generate_pdf - LOADED_INPUT - Loaded input "calculRJ" using input manager "io_manager", from output "result" of step "calculRJ"
[32m2025-04-08 23:35:15 +0200[0m - dagster - [34mDEBUG[0m - __ASSET_JOB - 9b56ea6b-712d-414e-96b4-5071e368c4e9 - generate_pdf - Loading file from: /home/rafik/Bureau/Data_Engineering/challengeData/rafik_pipeline/.tmp_dagster_home_hp2rsqei/storage/fetch_news using PickledObjectFilesystemIOManager...
[32m2025-04-08 23:35:15 +0200[0m - dagster - [34mDEBUG[0m - __ASSET_JOB - 9b56ea6b-712d-414e-96b4-5071e368c4e9 - 440340 - generate_pdf - LOADED_INPUT - Loaded input "fetch_news" using input manager "io_manager", from output "result" of step "fetch_news"
[32m2025-04-08 23:35:15 +0200[0m - dagster - [34mDEBUG[0m - __ASSET_JOB - 9b56ea6b-712d-414e-96b4-5071e368c4e9 - 440340 - generate_pdf - STEP_INPUT - Got input "calculRJ" of type "DataFrame". (Type check passed).
[32m2025-04-08 23:35:15 +0200[0m - dagster - [34mDEBUG[0m - __ASSET_JOB - 9b56ea6b-712d-414e-96b4-5071e368c4e9 - 440340 - generate_pdf - STEP_INPUT - Got input "fetch_news" of type "dict". (Type check passed).
[32m2025-04-08 23:35:15 +0200[0m - dagster - [34mERROR[0m - [31m__ASSET_JOB - 9b56ea6b-712d-414e-96b4-5071e368c4e9 - generate_pdf - Erreur lors de la g√©n√©ration du PDF: FPDF error: Undefined font: dejavu B[0m
[32m2025-04-08 23:35:15 +0200[0m - dagster - [34mERROR[0m - [31m__ASSET_JOB - 9b56ea6b-712d-414e-96b4-5071e368c4e9 - 440340 - generate_pdf - STEP_FAILURE - Execution of step "generate_pdf" failed.

dagster._core.errors.DagsterExecutionStepExecutionError: Error occurred while executing op "generate_pdf"::

RuntimeError: FPDF error: Undefined font: dejavu B

Stack Trace:
  File "/home/rafik/.local/lib/python3.12/site-packages/dagster/_core/execution/plan/utils.py", line 56, in op_execution_error_boundary
    yield
  File "/home/rafik/.local/lib/python3.12/site-packages/dagster/_utils/__init__.py", line 391, in iterate_with_context
    next_output = next(iterator)
                  ^^^^^^^^^^^^^^
  File "/home/rafik/.local/lib/python3.12/site-packages/dagster/_core/execution/plan/compute_generator.py", line 127, in _coerce_op_compute_fn_to_iterator
    result = invoke_compute_fn(
             ^^^^^^^^^^^^^^^^^^
  File "/home/rafik/.local/lib/python3.12/site-packages/dagster/_core/execution/plan/compute_generator.py", line 115, in invoke_compute_fn
    return fn(context, **args_to_pass) if context_arg_provided else fn(**args_to_pass)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/rafik/Bureau/Data_Engineering/challengeData/rafik_pipeline/rafik_pipeline/assets.py", line 148, in generate_pdf
    pdf.set_font("DejaVu", "B", 14)
  File "/home/rafik/.local/lib/python3.12/site-packages/fpdf/fpdf.py", line 603, in set_font
    self.error('Undefined font: '+family+' '+style)
  File "/home/rafik/.local/lib/python3.12/site-packages/fpdf/fpdf.py", line 227, in error
    raise RuntimeError('FPDF error: '+msg)
[0m
